backend/isisInternal.py

```diff
@@ -246,6 +246,31 @@ def getSpectrumRange(instrument, cycle, runs):
     data = nexusInteraction.getSpectrumRange(instrument, cycle, runs)
     return jsonify(data)
 
+# Check for data modifications
+
+
+@app.route('/pingCycle/<instrument>/<cycle>')
+def pingCycle(instrument, cycle):
+    url = 'http://data.isis.rl.ac.uk/journals/ndx'
+    url += "nimrod"+'/journal_main.xml'
+    response = urlopen(url)
+    tree = parse(response)
+    root = tree.getroot()
+    recentJournal = root[-1].get('name')
+    if (recentJournal != cycle):
+        return("New Cycle")
+
+    url = 'http://data.isis.rl.ac.uk/journals/ndx'
+    url += instrument + "/" + cycle
+    response = urlopen(url)
+    lastModified = response.info().get('Last-Modified')
+    lastModified = datetime.strptime(lastModified, "%a, %d %b %Y %H:%M:%S %Z")
+    currentTime = datetime.now()
+    timeDelta = currentTime - lastModified
+    if (timeDelta.total_seconds() < 300):
```

Comment:
The 300 seconds hardcoded here is of course related to the `QTimer` interval, but this seems a bit disconnected. Perhaps a neater way would be to store the modification time string when the journal(s) get retrieved in the first place - then, this function just needs to to a straight string comparison of the modification times to know whether the journal has been modified.  Catering for the appearance of a new cycle could be done in the same way on the journal index file.